tsdb/head_read.go

```diff
@@ -498,6 +501,8 @@ func (it *memSafeIterator) At() (int64, float64) {
 	return s.t, s.v
 }
 
+// stopIterator wraps an Iterator, but only returns the first
+// stopAfter-1 values. // TODO should this not be after stopAfter ?
```

Comment:
It is indeed after stopAfter. `i` is initialised to -1 above. A newStopIterator() can make it more easily understandable in case you want to add, but out of scope of this PR .
```suggestion
// stopIterator wraps an Iterator, but only returns the first
// stopAfter values.
```

Reply:
ah i see. it honors stopAfter if initialized with i=-1 (i assumed it would be initialized with the default zero value). I'll add a note for this.